From 71ec4cb2071eb819b814f7f7c6a6869a89e7e91e Mon Sep 17 00:00:00 2001
From: Pierre-Hugues Husson <phh@phh.me>
Date: Wed, 29 Apr 2020 00:17:01 +0200
Subject: [PATCH 18/24] Revert "Ignore unknown audio codecs in media profiles'
 AudioEncoder Cap"

This reverts commit ee00227dc5ddfc88b13fa37ba96f846950c03b4e.
---
 media/libmedia/MediaProfiles.cpp | 5 +----
 1 file changed, 1 insertion(+), 4 deletions(-)

diff --git a/media/libmedia/MediaProfiles.cpp b/media/libmedia/MediaProfiles.cpp
index 08d0199a07..98c54971ad 100644
--- a/media/libmedia/MediaProfiles.cpp
+++ b/media/libmedia/MediaProfiles.cpp
@@ -300,7 +300,6 @@ MediaProfiles::createAudioEncoderCap(const char **atts)
 
     const size_t nMappings = sizeof(sAudioEncoderNameMap)/sizeof(sAudioEncoderNameMap[0]);
     const int codec = findTagForName(sAudioEncoderNameMap, nMappings, atts[1]);
-    if(codec == -1) return nullptr;
     CHECK(codec != -1);
 
     MediaProfiles::AudioEncoderCap *cap =
@@ -420,9 +419,7 @@ MediaProfiles::startElementHandler(void *userData, const char *name, const char
         profiles->mVideoEncoders.add(createVideoEncoderCap(atts));
     } else if (strcmp("AudioEncoderCap", name) == 0 &&
                strcmp("true", atts[3]) == 0) {
-	    MediaProfiles::AudioEncoderCap* cap = createAudioEncoderCap(atts);
-        if(cap != nullptr)
-            profiles->mAudioEncoders.add(cap);
+        profiles->mAudioEncoders.add(createAudioEncoderCap(atts));
     } else if (strcmp("VideoDecoderCap", name) == 0 &&
                strcmp("true", atts[3]) == 0) {
         profiles->mVideoDecoders.add(createVideoDecoderCap(atts));
-- 
2.17.1

